llama_stack/apis/agents/agents.py

```diff
@@ -180,7 +180,8 @@ class Session(BaseModel):
 
     session_id: str
     session_name: str
-    turns: List[Turn]
+    turns: Optional[List[Turn]] = Field(default_factory=list)
```

Comment:
Why change this to optional?

Reply:
Because in the impl, I was hoping to merge this class with Session

```
class AgentSessionInfo(BaseModel):
    session_id: str
    session_name: str
    # TODO: is this used anywhere?
    vector_db_id: Optional[str] = None
    started_at: datetime
```

Since it duplicates a lot from the parent class Session.

Reply:
And `turns` is not present during the creation of the Session.

Reply:
> Because in the impl, I was hoping to merge this class with Session (AgentSessionInfo)

Ah yes, `AgentSessionInfo` is a implementation specific pydantic class that is not exposed to the API. We should plan on unify and get rid of it, but it is hard to do currently b/c adhoc implementation depends on `vector_db_id` which we shouldn't have in a Session.

Reply:
So, are you okay with making `turn` optional? :)

Reply:
I think we can keep `List[Turn]` and just return an empty list if there's no turns to keep things simple?